app/logic/registration/services/registrationServices.ts

```diff
@@ -0,0 +1,15 @@
+interface RegistrationInput {
+    email: string,
+    password: string
+}
+
+export class RegistrationServicesImpl {
+    public static checkInputForApi = ({email, password}: RegistrationInput): boolean => {
+        if (!(password.length >= 8)) return false
+        if (!(/[A-Z]/.test(password))) return false
+        const emailRegex = /^\s*[\w\-\+_]+(\.[\w\-\+_]+)*\@[\w\-\+_]+\.[\w\-\+_]+(\.[\w\-\+_]+)*\s*$/
+        if (!(emailRegex.test(email))) return false
+        const specialCharRegex = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/
+        return specialCharRegex.test(password);
```

Comment:
if ( || || || || || )

should only process the first one that return false therefore its the same as above but with less character (but const must be assigned before so its a choice to make) @julesgabriel

Reply:
why not make an if() with multiple parameter with if( parameterA || parameterB || etc)? it's maybe a dumb question for you

Reply:
Ok so here @alexandre1811 we are applying what we call 'early exit' which means we try to return the false before anything else, since it's easier to verify whats wrong than whats right

which means that by doing if on false statement we can return before doing anything else, here the order is based on the complexity of the statement